index.js

```diff
@@ -42,13 +52,39 @@ exports.register = (server, options, next) => {
                 if(!route.config.validate) {
                     route.config.validate = {};
                 }
-                
+
                 route.config.validate.headers = Joi.object({
                         Authorization: Joi.string().description('JWT token')
                     }).unknown();
             }
 
             server.route(route);
+            if(route.handler) {
+                if(getParamNames(route.handler).length < 2) {
+                    // default signature is (request, reply)
+                    // if reply is omitted, then the return value should be interpreted
+                    // as the reply.
+                    // we wrap the handler function to support this
+                    let orignalHandler = route.handler;
+                    route.handler = (request, reply) => {
+                        let retValue = orignalHandler.call(request, reply);
+                        if(retValue instanceof Promise) {
+                          return retValue
+                          .then(reply)
+                          .catch(err => {
+                              // handle promise rejection
+                              server.log(['startup', 'route-load', 'error'], err);
```

Comment:
This would not be a startup error if Promise rejects.

Maybe `request.log([request.method, request.path, 'error'], JSON.stringify(err));` ?